tests/unit/BusinessName.spec.ts

```diff
@@ -20,142 +20,157 @@ const store = getVuexStore()
+  }
+
+  function expectCorrectNameExists (exists = true): void {
+    expect(wrapper.findComponent(CorrectName).exists()).toBe(exists)
+  }
+
+  function expectNameRequestInfoExists (exists = true): void {
+    expect(wrapper.findComponent(NameRequestInfo).exists()).toBe(exists)
+  }
+
+  function expectUndoButtonExists (exists = true): void {
+    expect(wrapper.find('.btn-undo').exists()).toBe(exists)
+    if (exists) expect(wrapper.find('.btn-undo span').text()).toBe('Undo')
+  }
+
+  beforeEach(() => {
+    // tombstone data
     store.state.stateModel.businessId = 'BC1234567'
     store.state.stateModel.entityType = 'BC'
     store.state.stateModel.business.legalName = 'My Business Name'
-    store.state.stateModel.nameRequestApprovedName = 'NR Approved Name'
     store.state.stateModel.tombstone.filingType = 'restoration'
 
-    wrapperFactory = () => shallowMount(BusinessName, {
-      store,
-      vuetify
-    })
-  })
-
-  beforeEach(() => {
-    // empty restoration before each test
-    store.state.stateModel.restoration = {
-      applicationDate: null,
-      approvalType: null,
-      approvalTypeValid: true,
-      businessNameValid: false,
-      courtOrder: {
-        fileNumber: null
-      },
-      expiry: null,
-      noticeDate: null,
-      relationships: [],
-      restorationTypeValid: false,
-      type: null
-    }
-
-    // empty name request before each test
-    store.state.stateModel.nameRequest = {
-      applicants: {},
-      consentFlag: null,
-      expirationDate: null,
-      furnished: null,
-      legalType: null,
-      names: [],
-      nrNum: '',
-      priorityCd: null,
-      requestTypeCd: null,
-      request_action_cd: null,
-      state: null
-    }
-
-    // initial values before each test
+    // initial values
+    store.state.stateModel.correctNameOption = null
     store.state.stateModel.nameRequestApprovedName = null
     store.state.stateModel.showErrors = false
-  })
 
-  it('renders the component', () => {
-    const wrapper = wrapperFactory()
-    expect(wrapper.find('#business-name').exists()).toBe(true)
+    wrapper = shallowMount(BusinessName, { store, vuetify })
   })
 
-  it('displays editing mode sub-components', () => {
-    const wrapper = wrapperFactory()
-    expect(wrapper.find('label').text()).toBe('Business Name')
-    expect(wrapper.findComponent(CorrectName).exists()).toBe(true)
+  afterEach(() => {
+    wrapper.destroy()
   })
 
-  it('displays display mode sub-components', () => {
-    store.state.stateModel.nameRequestApprovedName = 'NR Approved Name'
-
-    const wrapper = wrapperFactory()
-    expect(wrapper.findComponent(NameRequestInfo).exists()).toBe(true)
+  it('renders the component in editing mode initially', () => {
+    expectBusinessNameExists()
+    expectSectionExists()
+    expectSectionValid()
+    expectLabelExists()
+    expectCorrectNameExists()
+    expectNameRequestInfoExists(false)
+    expectUndoButtonExists(false)
```

Comment:
I thought I'd try something new to keep the repetition down :)

Reply:
I like this! Not everything is crammed in here.

Reply:
Way to go!  Much more readable!!